### PR TITLE
ajax: documented new post and get config object interfaces

### PR DESCRIPTION
**Pull request by togakangaroo**
Documents changes in PR [1999](https://github.com/jquery/jquery/issues/1999)

### CODE REVIEW

**Comment by togakangaroo** [comment]
Unfortunately I couldn't get the vagrant VM running so this is untested but I think it's alright?

File: entries/jQuery.get.xml [hunk 1]
```diff
@@ -2,6 +2,12 @@
 <entry type="method" name="jQuery.get" return="jqXHR">
   <title>jQuery.get()</title>
   <signature>
+    <added>2.2</added>
+    <argument name="settings" type="PlainObject" optional="false">
+      <desc>A set of key/value pairs that configure the Ajax request. All properties except for <code>url</code> are optional. A default can be set for any option with <a href="/jQuery.ajaxSetup/">$.ajaxSetup()</a>. See <a href="/jquery.ajax/#jQuery-ajax-settings">jQuery.ajax( settings )</a> for a complete list of all settings. Type will automatically be set to <code>POST</code></desc>
```
Comment (arthurvr) [inline entries/jQuery.get.xml]: Missing dot at the end of your sentence.

File: entries/jQuery.post.xml [hunk 2]
```diff
@@ -2,6 +2,12 @@
 <entry type="method" name="jQuery.post" return="jqXHR">
   <title>jQuery.post()</title>
   <signature>
+    <added>2.2</added>
+    <argument name="settings" type="PlainObject" optional="false">
+      <desc>A set of key/value pairs that configure the Ajax request. All properties except for <code>url</code> are optional. A default can be set for any option with <a href="/jQuery.ajaxSetup/">$.ajaxSetup()</a>. See <a href="/jquery.ajax/#jQuery-ajax-settings">jQuery.ajax( settings )</a> for a complete list of all settings. Type will automatically be set to <code>POST</code></desc>
```
Comment (arthurvr) [inline entries/jQuery.post.xml]: Same, missing dot here.

File: entries/jQuery.get.xml (hunk 1 above)
Comment (arthurvr) [inline entries/jQuery.get.xml]: > Type will automatically be set to <code>POST</code></desc>

This seems more clear to me: 

> The `type` option will automatically be set to `POST`.

**Comment by arthurvr** [comment]
For the reference, issue is https://github.com/jquery/api.jquery.com/issues/620.

----------------------------

IMHO I'd like to see an example using the new technique in here.

> Unfortunately I couldn't get the vagrant VM running so this is untested but I think it's alright?

@togakangaroo Looks alright.

> Documents changes in PR 1999

You might mean https://github.com/jquery/jquery/pull/1995 ;)

Thanks for the PR! It's awesome when people doing stuff in core come in here and update the docs! :smile:

File: entries/jQuery.get.xml [hunk 3]
```diff
@@ -2,6 +2,12 @@
 <entry type="method" name="jQuery.get" return="jqXHR">
   <title>jQuery.get()</title>
   <signature>
+    <added>2.2</added>
```
Comment (dmethvin) [inline entries/jQuery.get.xml]: This will end up being `3.0`, that's our next release.

File: entries/jQuery.get.xml (hunk 1 above)
Comment (dmethvin) [inline entries/jQuery.get.xml]: Awesome that you caught the `$.ajaxSetup` point.

Comment (dmethvin) [inline entries/jQuery.get.xml]: I think we should be consistent within this paragraph, right now it's `$.ajaxSetup` and `jQuery.ajax`. I know we use both in docs but whichever is most common elsewhere should be used.

File: entries/jQuery.post.xml (hunk 2 above)
Comment (dmethvin) [inline entries/jQuery.post.xml]: Also same point regarding consistency.

**Comment by dmethvin** [comment]
@arthurvr do you have a working docs setup? Can you help @togakangaroo by pulling down this PR and seeing if it passes XMLLint checks?

**Comment by arthurvr** [comment]
@dmethvin That's what I meant with "looks alright". Everything passes and is building/deploying nicely.

**Comment by togakangaroo** [comment]
Alright, fixed up the points above. @dmethvin regarding `jQuery.ajax` vs `$.ajaxSetup`, sounds like something that is bigger than this PR, right? Or do you want me to take a look at it here?

**Comment by dmethvin** [comment]
Let's just be consistent here for both. Based on the `jQuery.ajax` entry I think we should use `$.get`, `$.ajaxSetup`, etc. inside paragraphs.

File: entries/jQuery.get.xml [hunk 4]
```diff
@@ -2,6 +2,12 @@
 <entry type="method" name="jQuery.get" return="jqXHR">
   <title>jQuery.get()</title>
   <signature>
+    <added>3.0</added>
+    <argument name="settings" type="PlainObject" optional="false">
+      <desc>A set of key/value pairs that configure the Ajax request. All properties except for <code>url</code> are optional. A default can be set for any option with <a href="/jQuery.ajaxSetup/">$.ajaxSetup()</a>. See <a href="/jquery.ajax/#jQuery-ajax-settings">jQuery.ajax( settings )</a> for a complete list of all settings. The type option will automatically be set to <code>POST</code>.</desc>
+    </argument>
+  </signature>
+   <signature>
```
Comment (arthurvr) [inline entries/jQuery.get.xml]: Looks like this breaks the indentation. Could you fix that? I'd love to land this PR soon.

Comment (dmethvin) [inline entries/jQuery.get.xml]: Arthur, you could edit this with an `--amend` IMO, it's a pretty trivial thing.

Comment (arthurvr) [inline entries/jQuery.get.xml]: Perfect. Landing right now.

**Pull request closed without merging**